pkg/policy/rule_test.go

```diff
@@ -419,29 +420,32 @@ func (ds *PolicyTestSuite) TestMergeL4PolicyEgress(c *C) {
 		},
 	}
 
-	mergedES := []api.EndpointSelector{fooSelector, bazSelector}
+	mergedES := CachedSelectorSlice{cachedFooSelector, cachedBazSelector}
 	expected := NewL4Policy()
-	expected.Egress["80/TCP"] = L4Filter{
-		Port: 80, Protocol: api.ProtoTCP, U8Proto: 6, Endpoints: mergedES,
+	expected.Egress["80/TCP"] = &L4Filter{
+		Port: 80, Protocol: api.ProtoTCP, U8Proto: 6, CachedSelectors: mergedES,
 		L7Parser: ParserTypeNone, L7RulesPerEp: L7DataMap{}, Ingress: false,
 		DerivedFromRules: labels.LabelArrayList{nil, nil},
 	}
 
 	state := traceState{}
-	res, err := rule1.resolveEgressPolicy(fromBar, &state, NewL4Policy(), nil)
+	res, err := rule1.resolveEgressPolicy(fromBar, &state, NewL4Policy(), nil, testSelectorCache)
+
+	c.Log(buffer)
+
 	c.Assert(err, IsNil)
 	c.Assert(res, Not(IsNil))
-	c.Assert(*res, checker.DeepEquals, *expected)
+	c.Assert(*res, checker.Equals, *expected)
```

Comment:
I think that all of the `DeepEquals` -> `Equals` changes could be split out to reduce the size of this patch.

Reply:
Will do if more rounds are needed.